configuration/menu.lua

```diff
@@ -225,15 +224,15 @@ local function reboot()
 end
 
 local function generateMenu()
-    selector = activator(location.x, location.y, selectionBoxWidth, maxheight-5, "Activate", activate, modules, "Available", selector)
-    deselector = activator(location.x+selectionBoxWidth+1, location.y, selectionBoxWidth, maxheight-5, "Disable", deactivate, processes, "Active", deselector)
-    gui.bigButton(location.x, location.y+maxheight-5, "Run", switchRunStatus)
-    gui.bigButton(location.x+5, location.y+maxheight-5, "Save", save)
-    gui.bigButton(location.x+11, location.y+maxheight-5, "Reboot", reboot)
-    gui.bigButton(location.x+19, location.y+maxheight-5, "Shell", interrupt)
-    gui.bigButton(location.x+26, location.y+maxheight-5, "Settings", configScreen, {location.x+2*selectionBoxWidth+2, location.y, maxWidth-(location.x+2*selectionBoxWidth+2), maxheight-5, "NIDAS Settings", menu})
-    if updateAvailable() then gui.smallButton(maxWidth-21, maxheight, "Update available!", update) end
-    gui.smallLogo(maxWidth-20, maxheight-4, require("nidas_version"))
+    selector = activator(location.x, location.y, selectionBoxWidth, graphics.context().height-5, "Activate", activate, modules, "Available", selector)
+    deselector = activator(location.x+selectionBoxWidth+1, location.y, selectionBoxWidth, graphics.context().height-5, "Disable", deactivate, processes, "Active", deselector)
+    gui.bigButton(location.x, location.y+graphics.context().height-5, "Run", switchRunStatus)
+    gui.bigButton(location.x+5, location.y+graphics.context().height-5, "Save", save)
+    gui.bigButton(location.x+11, location.y+graphics.context().height-5, "Reboot", reboot)
+    gui.bigButton(location.x+19, location.y+graphics.context().height-5, "Shell", interrupt)
+    gui.bigButton(location.x+26, location.y+graphics.context().height-5, "Settings", configScreen, {location.x+2*selectionBoxWidth+2, location.y, graphics.context().width-(location.x+2*selectionBoxWidth+2), graphics.context().height-5, "NIDAS Settings", menu})
+    if updateAvailable() then gui.smallButton(graphics.context().width-21, graphics.context().height, "Update available!", update) end
+    gui.smallLogo(graphics.context().width-20, graphics.context().height-4, require("nidas_version"))
     component.gpu.fill(1, 1, 160, 50, " ")
     renderer.update()
 end
```

Comment:
Calls take a while, could put them all in a variable

Reply:
True
I just find&replaced the old variable with the call.

Reply:
Bad practice